litgpt/finetune/adapter.py

```diff
@@ -294,13 +291,20 @@ def fit(
                 save_hyperparameters(setup, checkpoint_file.parent)
                 save_prompt_style(data.prompt_style, checkpoint_file.parent)
 
+        if step_count == max_steps:
+            fabric.print("Final validation ...")
+            train_loss = validate(fabric, model, train_dataloader, tokenizer, eval, data, print_output=False)
```

Comment:
This is slow. Why not rely on the running loss that is already printed?

Reply:
> This is slow. Why not rely on the running loss that is already printed?

I agree, this it is slow, but it's "only" exactly 2 times as slow as running validate as before. Sure, we could use the training loss from the previous iteration but then you have the same issue you pointed out for the validation loss:

> Notice that this only does 2 iters of validation. Showing a value here could be very misleading.

In practice it is useful to compare training and validation loss fairly to know the degree of overfitting in my opinion. We can maybe add a flag: `skip_validation` with default to `False` when we are running speed benchmarks. But for real uses cases, I think that knowing the training and validation losses is important.